src/main/java/org/jenkinsci/plugins/ssegateway/Endpoint.java

```diff
@@ -100,6 +100,24 @@ protected void init() throws ServletException {
         PluginServletFilter.addFilter(new SSEListenChannelFilter());
     }
 
+    /**
+     * Returns the root directory for sse-events based on the Jenkins logs directory
+     * (historically always found under <code>$JENKINS_HOME/logs</code>.
+     * Configurable since Jenkins 2.114.
+     *
+     * @see hudson.triggers.SafeTimerTask#LOGS_ROOT_PATH_PROPERTY
+     * @return the root directory for logs.
+     */
+    private File getHistoryDirectory(Jenkins jenkins) {
+        final String overriddenLogsRoot = System.getProperty("hudson.triggers.SafeTimerTask.logsTargetDir");
```

Comment:
Seems unnecessarily tied to an implementation detail of core.

Reply:
@daniel-beck what do you mean? That those files should be put elsewhere? 

I think the core should actually maybe even more define an API for those things to avoid putting random files everywhere BTW. Cf. also the ongoing initiative about Jenkins logging in general.

Reply:
> what do you mean?

You don't check a core API to make this decision, but the mechanism by which core (currently) decides to put those files elsewhere.